### PR TITLE
feat: adding first page for screen recording 

### PR DESCRIPTION
**Pull request by Swapnilden**
- Added a new page `screen-recorder` for screen recording
 - "Record Video" button added
- on clicking the record video buttont it creates a video recording, shows a preview and allows download

### CODE REVIEW

File: frontend/webapp/src/pages/index.js [hunk 1]
```diff
@@ -17,15 +17,24 @@ export default function Home() {
...
-      <main className="flex flex-col gap-8 row-start-2">
-        + Record Video
+
+      {/* Main Content */}
+      <main className="flex flex-col gap-8 row-start-2 items-center">
+        <button
+          className="px-6 py-3 bg-blue-600 text-white rounded-lg hover:bg-blue-700 transition-all"
+          onClick={() => {
+            // You can navigate to /screen-recorder or trigger functionality here
+            window.location.href = "/screen-recorder";
```
Comment (xask00) [inline frontend/webapp/src/pages/index.js:31]: Please use the 'react' or 'nextjs' wat to navigate to a new page,

File: frontend/webapp/src/pages/screen-recorder.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
+import { useState, useRef } from "react";
+
+export default function ScreenRecorder() {
+  const [isRecording, setIsRecording] = useState(false);
+  const [videoUrl, setVideoUrl] = useState(null);
+  const mediaRecorder = useRef(null);
+  const chunks = useRef([]);
+
+  const startRecording = async () => {
```
Comment (xask00) [inline frontend/webapp/src/pages/screen-recorder.js:9]: on testing this we dont hear the audio getting recorded, please verify that audio also gets recorded

**Review by xask00 — CHANGES_REQUESTED** [review]

File: frontend/webapp/src/pages/screen-recorder.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    try {
+      // Request screen recording permission
+      const stream = await navigator.mediaDevices.getDisplayMedia({
```
Comment (xask00) [inline frontend/webapp/src/pages/screen-recorder.js:12]: even after the recording stops, the browser url window still has a 'recording' icon and shows "this tab is sharing your screen'

File: frontend/webapp/src/pages/screen-recorder.js (hunk 2 above)
Comment (Swapnilden) [reply frontend/webapp/src/pages/screen-recorder.js:9]: sir it is recording audio only for window

File: frontend/webapp/src/pages/screen-recorder.js (hunk 3 above)
Comment (Swapnilden) [reply frontend/webapp/src/pages/screen-recorder.js:12]: i will try to solve it sir

File: frontend/webapp/src/pages/screen-recorder.js (hunk 2 above)
Comment (xask00) [reply frontend/webapp/src/pages/screen-recorder.js:9]: this is not resolved,
make sure it is recording audio by default, so we do not have to manually toggle the audio button everttime

Comment (xask00) [reply frontend/webapp/src/pages/screen-recorder.js:9]: attach screenshots, short video for showing us what you mean

File: frontend/webapp/src/pages/screen-recorder.js (hunk 3 above)
Comment (xask00) [reply frontend/webapp/src/pages/screen-recorder.js:12]: no this is not what we are expecting,
when we click on stop recording, it should stop sharing automatically, we should not need to do it manually.

Comment (Swapnilden) [reply frontend/webapp/src/pages/screen-recorder.js:10]: okay sir let me enhance the code

File: frontend/webapp/src/pages/index.js [hunk 4]
```diff
@@ -17,15 +17,24 @@ export default function Home() {
     <div
       className={`${geistSans.variable} ${geistMono.variable} grid grid-rows-[20px_1fr_20px] items-center justify-items-center min-h-screen p-8 pb-20 gap-16 sm:p-20 font-[family-name:var(--font-geist-sans)]`}
```
Comment (janakiramg) [inline frontend/webapp/src/pages/index.js:18]: @Swapnilden Please remove backed folder(including go.work.sum fule). We don't need it for this initial task

File: frontend/webapp/src/pages/index.js (hunk 1 above)
Comment (janakiramg) [reply frontend/webapp/src/pages/index.js:31]: @Swapnilden  Can you please address this review comment ?

File: frontend/webapp/src/pages/screen-recorder.js (hunk 2 above)
Comment (janakiramg) [reply frontend/webapp/src/pages/screen-recorder.js:9]: @Swapnilden Did you get a chance to to fix audio ?

File: frontend/webapp/src/pages/screen-recorder.js [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+export default function ScreenRecorder() {
+  const [isRecording, setIsRecording] = useState(false);
+  const [videoUrl, setVideoUrl] = useState(null);
+  const mediaRecorder = useRef(null);
+  const recordedChunks = useRef([]);
+
+  const startRecording = async () => {
+    try {
+      // Request screen recording
+      const screenStream = await navigator.mediaDevices.getDisplayMedia({
```
Comment (janakiramg) [inline frontend/webapp/src/pages/screen-recorder.js:12]: @Swapnilden Did you fix below issue ?
 "even after the recording stops, the browser url window still has a 'recording' icon and shows "this tab is sharing your screen' "

**Review by janakiramg — CHANGES_REQUESTED** [review]

File: frontend/webapp/src/pages/screen-recorder.js (hunk 2 above)
Comment (Swapnilden) [reply frontend/webapp/src/pages/screen-recorder.js:9]: @janakiramg Yes, the audio issue was addressed

File: frontend/webapp/src/pages/screen-recorder.js (hunk 5 above)
Comment (Swapnilden) [reply frontend/webapp/src/pages/screen-recorder.js:12]: @janakiramg yes

File: frontend/webapp/src/pages/screen-recorder.js (hunk 2 above)
Comment (xask00) [reply frontend/webapp/src/pages/screen-recorder.js:9]: I can verify from the demo video that audio is being recorded and played back

**Pull request merged**